### DISCUSSION TITLE
Block editor ctrl-u (insert line) does not work properly?

### DISCUSSION THREAD

**Discussion by mcctuxic** [tabemann/zeptoforth#163] [Q&A]
Hi,

using the following software
Gentoo-Linux (last update this morning)
picocom 3.1
with eiher
wezterm-20240203.110809-r2
kitty 0.37.0
alacritty 0.14.0-r1
xterm 395 (out of curiousity)
Zeptoforth 1.9.0
on a Pi Pico 2 W with RP2040
and
on a Pi Pico 2 W with RP2350
Using the "raw" full-usb.uf2 image of zeptoforth and the state of the machine right after reboot.

command used 

10 edit

Block 10 has some forth code - beside others line 1 is fille with some code

Positioning the cursor at line 1 and pressing CTRL-U results in:
no reaction at all on the RP2040 boards and
one empty line inserted above line 1 on the RP2350 board
Addtional CTRL-Us are ignored by both boards.

I tried that four terminal emulators with the same result.

Picocom uses CTRL-A as a prefix to all commands so I think, picocom
cannnot be the reason.

Where can I look further to make CTRL-U working?

Cheers!
mcc

**Comment by tabemann** [comment]
On Fri, Dec 6, 2024 at 3:04 AM mcctuxic ***@***.***> wrote:

> Hi,
>
> using the following software
> Gentoo-Linux (last update this morning)
> picocom 3.1
> with eiher
> wezterm-20240203.110809-r2
> kitty 0.37.0
> alacritty 0.14.0-r1
> xterm 395 (out of curiousity)
> Zeptoforth 1.9.0
> on a Pi Pico 2 W with RP2040
> and
> on a Pi Pico 2 W with RP2350
> Using the "raw" full-usb.uf2 image of zeptoforth and the state of the
> machine right after reboot.
>
> command used
>
> 10 edit
>
> Block 10 has some forth code - beside others line 1 is fille with some code
>
> Positioning the cursor at line 1 and pressing CTRL-U results in:
> no reaction at all on the RP2040 boards and
> one empty line inserted above line 1 on the RP2350 board
> Addtional CTRL-Us are ignored by both boards.
>
> I tried that four terminal emulators with the same result.
>
> Picocom uses CTRL-A as a prefix to all commands so I think, picocom
> cannnot be the reason.
>
> Where can I look further to make CTRL-U working?
>
> Cheers!
> mcc
>
> I was unable to reproduce this issue on either the Raspberry Pi Pico or
the Raspberry Pi Pico 2 W (W versus non-W does not affect this).

I first completely erased all of flash with a recent version of
flash_nuke.uf2 (a version that works on both RP2040 and RP2350 boards). I
then installed bin/1.9.0/rp2040_big/zeptoforth_full_usb-1.9.0.uf2 on the
former and bin/1.9.0/rp2350/zeptoforth_full_usb-1.9.0.uf2 on the latter.

I executed 10 EDIT on each of them. Control-U behaved as expected on both
-- I entered a few 'z's on the first line, then pressed control-U a few
times and a few blank lines were inserted. Then I pressed control-X
followed by control-V to revert my changes and then exit out without
writing to flash. Then I repeated the same, except I entered control-A
control-A (I was using picocom) to return to the start of the line after
enter the 'z's and before entering the control-U's, and it behaved the same
way except this time the cursor was at the start of the line.

I repeated this a number of times on each of them, with the same results
each time.

I really don't know what is going wrong for you. The only thing I can think
of is is that the block storage above dictionary flash is corrupt on your
boards, and this is confusing EDIT (even though I don't think it should).
Try finding flash_nuke.uf2 and using it on your boards (unless you really
care about what is in block storage), and using it before then installing
the 1.9.0 full_usb UF2 files.

Travis

**Comment by mcctuxic** [comment]
On 12/06 04:11, tabemann wrote:
> On Fri, Dec 6, 2024 at 3:04 AM mcctuxic ***@***.***> wrote:
> 
> > Hi,
> >
> > using the following software
> > Gentoo-Linux (last update this morning)
> > picocom 3.1
> > with eiher
> > wezterm-20240203.110809-r2
> > kitty 0.37.0
> > alacritty 0.14.0-r1
> > xterm 395 (out of curiousity)
> > Zeptoforth 1.9.0
> > on a Pi Pico 2 W with RP2040
> > and
> > on a Pi Pico 2 W with RP2350
> > Using the "raw" full-usb.uf2 image of zeptoforth and the state of the
> > machine right after reboot.
> >
> > command used
> >
> > 10 edit
> >
> > Block 10 has some forth code - beside others line 1 is fille with some code
> >
> > Positioning the cursor at line 1 and pressing CTRL-U results in:
> > no reaction at all on the RP2040 boards and
> > one empty line inserted above line 1 on the RP2350 board
> > Addtional CTRL-Us are ignored by both boards.
> >
> > I tried that four terminal emulators with the same result.
> >
> > Picocom uses CTRL-A as a prefix to all commands so I think, picocom
> > cannnot be the reason.
> >
> > Where can I look further to make CTRL-U working?
> >
> > Cheers!
> > mcc
> >
> > I was unable to reproduce this issue on either the Raspberry Pi Pico or
> the Raspberry Pi Pico 2 W (W versus non-W does not affect this).
> 
> I first completely erased all of flash with a recent version of
> flash_nuke.uf2 (a version that works on both RP2040 and RP2350 boards). I
> then installed bin/1.9.0/rp2040_big/zeptoforth_full_usb-1.9.0.uf2 on the
> former and bin/1.9.0/rp2350/zeptoforth_full_usb-1.9.0.uf2 on the latter.
> 
> I executed 10 EDIT on each of them. Control-U behaved as expected on both
> -- I entered a few 'z's on the first line, then pressed control-U a few
> times and a few blank lines were inserted. Then I pressed control-X
> followed by control-V to revert my changes and then exit out without
> writing to flash. Then I repeated the same, except I entered control-A
> control-A (I was using picocom) to return to the start of the line after
> enter the 'z's and before entering the control-U's, and it behaved the same
> way except this time the cursor was at the start of the line.
> 
> I repeated this a number of times on each of them, with the same results
> each time.
> 
> I really don't know what is going wrong for you. The only thing I can think
> of is is that the block storage above dictionary flash is corrupt on your
> boards, and this is confusing EDIT (even though I don't think it should).
> Try finding flash_nuke.uf2 and using it on your boards (unless you really
> care about what is in block storage), and using it before then installing
> the 1.9.0 full_usb UF2 files.
> 
> Travis
> 
> > Message ID: ***@***.***>
> >

Hi Travis,

thanks for checking the effect.
I nuked my RS2350 board and after that, the problem vanished.
I will keep an eye on this effect - may be I will find a clue,
what may be the reason for this...

Cheers!
mcc
